### ISSUE TITLE
Plugin doesn't request new node due to incorrect currentDemand value

### ISSUE THREAD

**Issue by wallhided** [jenkinsci/ec2-fleet-plugin#359]
## Issue Details

**Describe the bug**
I'm trying to use a node for pipeline from existing fleet on ec2-instances.(I'm using MacOs instances)
Plugin doesn't launch new node. Perhaps currentDemand value was below 0.

Currently there are no nodes related to my label "mac-fleet". (in Jenkins and in AWS itself)

What I've found in logs:
```
label [mac-fleet]: currentDemand 0 availableCapacity 1 (availableExecutors 0 connectingExecutors 0 plannedCapacitySnapshot 1 additionalPlannedCapacity 0)
Jan 26, 2023 10:34:43 AM FINE com.amazon.jenkins.ec2fleet.NoDelayProvisionStrategy
label [mac-fleet]: currentDemand is less than 1, not provisioning
Jan 26, 2023 10:34:43 AM FINE com.amazon.jenkins.ec2fleet.NoDelayProvisionStrategy
Provisioning completed
```

After restarting of jenkins - issue disappears. Also if I start one more same pipeline, launch process starts. 

**To Reproduce**
1. Have consistent jenkins instance after restart
2. Create job with requesting node by label
3. Wait some time for node deprovision
4. Try step 2 one more time

## Environment Details

**Plugin Version?**
2.5.2

**Jenkins Version?**
2.361.1

**Spot Fleet or ASG?**
ASG of MacOs instances, Host resource group and etc are configured

**Label based fleet?**
Yes

**Linux or Windows?**
MacOs

**Anything else unique about your setup?**
MacOs ASG

**Comment by mtn-boblloyd** [comment]
I'm seeing a similar issue.  It seems that the LoadStatistics in Jenkins may not reflect Jenkins Pipeline queue data.  I've seen two cases present:
* Having a Spot Request with a TotalCapacity set to 0 causes no nodes to start if the first requesting job is a Pipeline step
* Having a Spot Request with a TotalCapacity set to 1 causes no _new_ nodes to start if there is one job running on a Spot Fleet node, and a second job, which is Pipeline step, starts.  In this case, it attaches to the running node and waits for an available executor instead of running a new node.

In both these cases, the Pipeline step that is waiting will wait either indefinitely (in the first case) or until the node executor is available (in the second case).  In both cases, the log message indicates that there is no `currentDemand` as the queue length is never seen to be > 0.

**Comment by mwos-sl** [comment]
Sounds similar to: https://github.com/jenkinsci/ec2-fleet-plugin/issues/322

**Comment by fruch** [comment]
the sad part, the only thing helps in this situation is to restart Jenkins.

I've seen it's happening mostly if some is manually scale up/down the fleet or ASG, and somehow the plugin get out of sync, or into negative numbers

**Comment by pdk27** [comment]
I believe this issue is fixed in [release 2.6.0](https://github.com/jenkinsci/ec2-fleet-plugin/releases/tag/ec2-fleet-2.6.0). Feel free to re-open with details / logs if the issue still exists.
